emsdk_env.sh

```diff
@@ -15,10 +15,25 @@
 #
 # which won't have any effect.
 if [ -z "$BASH_SOURCE" ]; then
-  if [ ! -f "./emsdk.py" ]; then
+  D1=`dirname $0`
+  DIR=`cd $D1 && pwd`
+  S=`basename $0`
+
+  READLINK=`which readlink`
+  if [ -x "$READLINK" ] ; then
+    DLINK=`readlink $D1/$S`
+    if [ "$DLINK" != "" ] ; then
+      D1=`dirname $DLINK`
+      DIR=`cd $D1 && pwd`
+    fi
+  fi
+  if [ ! -f "$DIR/emsdk.py" ]; then
     echo "error: You must be in the same directory as emsdk_env.sh when sourcing it (or switch to the bash shell)" 1>&2
```

Comment:
This error message could use some cleanup now perhaps.

How about: `error: unable to determine emsdk directory.  Perhaps you are using a shell or environment that this script does not support.  One possible solution is to source this script while in the emsdk directory.`

Reply:
Good idea on the error message. I kind of left it as is, but am happy to modify.